### PR TITLE
Enable use of 'select' and 'selectPrefix' options in CloudinaryImage type

### PR DESCRIPTION
**Pull request by azterix**
This is my fix to address the non-working 'select: true' option in the CloudinaryImage type - Issue #1340 

I modified Type.js to pull in options related to properties defined in subtypes to make the options available in this.props. I'm not 100% sure this is in line with the intent of what is expected of Types.js, so please clarify if I'm doing this wrong.

I added an array of properties in CloudinaryImageType.js that appear to be expected by the field template (select, selectPrefix, and autoCleanup).

I also updated the CloudinaryImageField.js to pass the selectPrefix option in to the cloudinary API in order to properly filter the available options.

### CODE REVIEW

**Comment by JedWatson** [comment]
This looks good to me, thanks @azterix!

Would you mind updating your PR so I can merge it? And please revert the change to package.json.

File: fields/types/Type.js [hunk 1]
```diff
@@ -99,6 +99,8 @@ Field.prototype.getOptions = function() {
 		optionKeys.forEach(function(key) {
 			if (this[key]) {
 				this.__options[key] = this[key];
+			} else if(this.options[key]){
```
Comment (JedWatson) [inline fields/types/Type.js]: should be a space here `if (this` and here `) {`

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "keystone",
-  "version": "0.3.10",
+  "version": "0.3.10.1",
```
Comment (JedWatson) [inline package.json]: Please don't change the version number, I do that when I prepare releases.

**Comment by czoellick-icf** [comment]
Updates posted. First time doing a pull request here - let me know if there's anything else that I need to do. Thanks!

**Comment by azterix** [comment]
The requested updates have been posted. First time doing a pull request here - let me know if there's anything else I need to do.

**Comment by Mentioum** [comment]
Is this going to be merged in? Specifically this commit? c2867611dc7205e6f5f7a7fa8edbd9c048cfab95

**Comment by morenoh149** [comment]
sadly the files have changed since. @azterix could you rebase against master again please.

**Comment by azterix** [comment]
Rebase completed

**Comment by JedWatson** [comment]
@azterix this reverts a number of important changes in `fields/types/Type.js` and includes a lot of whitespace changes which obscure the important differences in your PR. Would you mind fixing those issues? would be good to get this merged.

**Comment by azterix** [comment]
@JedWatson fixed.. apologies for the sloppy rebase.

**Comment by JedWatson** [comment]
No worries @azterix - thanks for the PR!

**Pull request merged**